pkg/api/types.go

```diff
@@ -396,6 +398,24 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts are Glusterfs hosts address array
+	Hosts []string `json:"hosts"`
```

Comment:
----- Original Message -----
> > @@ -396,6 +398,24 @@ type NFSVolumeSource struct {
> >  	ReadOnly bool `json:"readOnly,omitempty"`
> >  }
> >  
> > +// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the
> > lifetime of a pod
> > +type GlusterfsVolumeSource struct {
> > +	// Required: hosts are Glusterfs hosts address array
> > +	Hosts []string `json:"hosts"`
> 
> The dynamic DNS solution is good for services that expose 1-N name-to-IP
> mappings. This is however not ideal solution for glusterfs.
> - Operation complexity. The gluster cluster may not be in the same managed
> network domain. Sync-up DNS entries between gluster cluster and kube cluster
> adds complexity.

Today.  Probably not in the future.

> - Scaling. If you have a large glusterfs setup, scaling will be tremendous.
> - Performance. In a large kube setup, when all minions mount a gluster
> filesystem and they all query DNS for resolution, there is a good chance for
> something goes wrong.
> - Mount storm. If DNS servers reply to all the host the same order of gluster
> hosts, then minions could end up mounting the same gluster host, this is
> something I don't want to see.

With SkyDNS we are round-robining, but agree this could be an issue.

> - HA.  a DNS lookup may not be able to get all the entries. If a unfortunate
> query gets only a handful entries that happens to point to a unresponsive
> gluster hosts, mount then fail.

This ties into headless services exposed via DNS, but we have this working already in OpenShift and I'm fairly certain that it will scale to at least the cluster levels we're discussing now.

> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6174/files#r27501247